### DISCUSSION TITLE
csp

### DISCUSSION THREAD

**Discussion by inkss** [volantis-x/hexo-theme-volantis#737] [General]
现行 csp 不允许 new Function/eval，有什么解决方案吗？@MHuiG

**Comment by MHuiG** [comment]
new Function eval setInterval setTimeout 都被禁用了

**Reply by inkss** [reply]
![常见问题与反馈渠道 - Volantis 和另外 5 个页面 - 个人 - Microsoft​ E_22-04-11_15-38-27](https://user-images.githubusercontent.com/31947043/162687939-f5da83b9-4be2-4431-87ef-de1da3d6ec55.png)

**Reply by MHuiG** [reply]
这个图是 'unsafe-inline' 通常浏览器会报错提示应该怎么做

**Reply by inkss** [reply]
话说，为什么上面几个可以执行？

**Reply by MHuiG** [reply]
使用了 'unsafe-hashes' https://github.com/volantis-x/community/blob/main/gulpfile.js#L64

可以写个自动程序都提取出来 document.querySelectorAll("[onclick]").forEach(e=>{console.log(e.getAttribute("onclick"))})

**Reply by inkss** [reply]
学习了
